### PR TITLE
refactor: Move scalacli tests to a separate job

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we would run the Scala CLI tests together with normal unit tests, but that could cause OOM issues on the CI since we would efectively start 2 Bloop instances. Now, we run the ScalaCLI tests separately.

Also:
- updated tests to run in Scala 3
- updated the version of ScalaCLI

I did those together since there is an issue with scripts alongside scala files on Scala 2, but it doesn't make sense to block on this as the version we use is only used for single scripts (which works well) and anyway the users will most likely have the Scala CLI version installed locally.

### CODE REVIEW

File: tests/slow/src/main/scala/tests/scalacli/BaseScalaCliSuite.scala [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import scala.meta.internal.metals.Messages
+import scala.meta.internal.metals.scalacli.ScalaCli
+
+import ch.epfl.scala.bsp4j.MessageType
+import org.eclipse.lsp4j.InitializeResult
+import org.eclipse.lsp4j.MessageActionItem
+import tests.BaseLspSuite
+import tests.ScriptsAssertions
+
+abstract class BaseScalaCliSuite(protected val scalaVersion: String)
```
Comment (tgodzik) [inline tests/slow/src/main/scala/tests/scalacli/BaseScalaCliSuite.scala:22]: Ach and also I moved the tests from BaseScalaCLISuite as they were just run once anyway,

File: tests/slow/src/test/scala/tests/scalacli/ScalaCliSuite.scala [hunk 2]
```diff
@@ -301,13 +123,73 @@ abstract class BaseScalaCliSuite(scalaVersion: String)
...
+        |  }
+        |}
+        |
+        |/foo.sc
+        |class Foo {
+        |  def value = 5
+        |}
+        |""".stripMargin
+
+  test(s"simple-file-BSP") {
```
Comment (tgodzik) [inline tests/slow/src/test/scala/tests/scalacli/ScalaCliSuite.scala:153]: I changed these tests to just lsit them instead of a for comprehension.

**Comment by tgodzik** [comment]
Btw. everything was mostly moved, I just changed a tiny bit the ScalaCLISuite

**Review by ckipp01 — APPROVED** [review]
Looks good to me! Hopefully this will make CI less flaky in general. LGTM!

**Comment by tgodzik** [comment]
> Looks good to me! Hopefully this will make CI less flaky in general. LGTM!

That is my hope! :D

**Pull request merged**